synapse/storage/databases/main/transactions.py

```diff
@@ -164,7 +165,7 @@ def _get_destination_retry_timings(self, txn, destination):
             allow_none=True,
         )
 
-        if result and result["retry_last_ts"] > 0:
+        if result and result["retry_last_ts"] and result["retry_last_ts"] > 0:
```

Comment:
I'm slightly struggling to see how this is related. Can you help me out?

Reply:
`retry_last_ts` may be NULL — this was already the case before as far as the schema goes, but otherwise this is untrodden until now.

(This case is hit when we insert a `destinations` row to satisfy the constraint that `destination_rooms` has.)

Reply:
got it.

`result["retry_last_ts"]` can never be negative, so I'd omit the last clause.